hbase-server/src/test/java/org/apache/hadoop/hbase/master/assignment/TestTRSPPersistUninitializedSubProc.java

```diff
@@ -0,0 +1,125 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hbase.master.assignment;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import org.apache.hadoop.hbase.HBaseClassTestRule;
+import org.apache.hadoop.hbase.HBaseTestingUtil;
+import org.apache.hadoop.hbase.ServerName;
+import org.apache.hadoop.hbase.TableName;
+import org.apache.hadoop.hbase.client.RegionInfo;
+import org.apache.hadoop.hbase.master.HMaster;
+import org.apache.hadoop.hbase.master.assignment.TransitRegionStateProcedure.TransitionType;
+import org.apache.hadoop.hbase.master.procedure.MasterProcedureEnv;
+import org.apache.hadoop.hbase.master.procedure.ServerCrashProcedure;
+import org.apache.hadoop.hbase.procedure2.Procedure;
+import org.apache.hadoop.hbase.procedure2.ProcedureExecutor;
+import org.apache.hadoop.hbase.procedure2.ProcedureSuspendedException;
+import org.apache.hadoop.hbase.procedure2.ProcedureTestingUtility;
+import org.apache.hadoop.hbase.procedure2.ProcedureYieldException;
+import org.apache.hadoop.hbase.testclassification.MasterTests;
+import org.apache.hadoop.hbase.testclassification.MediumTests;
+import org.apache.hadoop.hbase.util.Bytes;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+import org.apache.hadoop.hbase.shaded.protobuf.generated.ProcedureProtos.ProcedureState;
+
+/**
+ * Testcase for HBASE-29259
+ */
+@Category({ MasterTests.class, MediumTests.class })
+public class TestTRSPPersistUninitializedSubProc {
+
+  @ClassRule
+  public static final HBaseClassTestRule CLASS_RULE =
+    HBaseClassTestRule.forClass(TestTRSPPersistUninitializedSubProc.class);
+
+  private static HBaseTestingUtil UTIL = new HBaseTestingUtil();
+
+  private static byte[] CF = Bytes.toBytes("cf");
+
+  private static TableName TN = TableName.valueOf("tn");
+
+  public static class TRSPForTest extends TransitRegionStateProcedure {
+
+    private boolean injected = false;
+
+    public TRSPForTest() {
+    }
+
+    public TRSPForTest(MasterProcedureEnv env, RegionInfo hri, ServerName assignCandidate,
+      boolean forceNewPlan, TransitionType type) {
+      super(env, hri, assignCandidate, forceNewPlan, type);
+    }
+
+    @Override
+    protected Procedure[] execute(MasterProcedureEnv env)
+      throws ProcedureSuspendedException, ProcedureYieldException, InterruptedException {
+      Procedure[] subProcs = super.execute(env);
+      if (!injected && subProcs != null && subProcs[0] instanceof CloseRegionProcedure) {
+        injected = true;
+        ServerName sn = ((CloseRegionProcedure) subProcs[0]).targetServer;
+        env.getMasterServices().getServerManager().expireServer(sn);
+        try {
+          UTIL.waitFor(15000, () -> env.getMasterServices().getProcedures().stream().anyMatch(
```

Comment:
Might as well make this retry for some long duration, like 5 mins?

Reply:
I think 15 seconds is enough, this is for triggering SCP, not for finishing SCP.